tornado/test/curl_httpclient_test.py

```diff
@@ -88,7 +90,8 @@ def setUp(self):
 
     def get_app(self):
         return Application([
-            ('/digest', DigestAuthHandler),
+            ('/digest', DigestAuthHandler, {'username': 'foo', 'password': 'bar'}),
+            ('/digest_non_ascii', DigestAuthHandler, {'username': 'foo', 'password': 'barユ£'}),
```

Comment:
The test should go in with the basic auth tests in `httpclient_test.py` instead of the digest auth tests here (both http clients support basic auth; only curl supports digest auth. Both basic auth implementations should support non-ascii usernames and passwords).

Include non-ascii characters in both the username and password - I tried adding that to this patch and it didn't work, although it didn't fail in the way I expected. I think something like @ploxiln's suggestion will be necessary.